### PR TITLE
Update to Swift 2

### PR DESCRIPTION
**Pull request by neilpa**

### CODE REVIEW

File: Commandant/Errors.swift [hunk 1]
```diff
@@ -7,29 +7,28 @@
...
 import Foundation
-import Box
 import Result
 
 /// Possible errors that can originate from Commandant.
 ///
 /// `ClientError` should be the type of error (if any) that can occur when
 /// running commands.
-public enum CommandantError<ClientError> {
+public enum CommandantError<ClientError>: ErrorType {
```
Comment (neilpa) [inline Commandant/Errors.swift]: Should we also enforce `ClientError` conform to `ErrorType`?

**Comment by jpsim** [comment]
This is neat, @neilpa! There's already a Swift 2 branch at `swift-2.0` (originally PR #26). These branches do a few things differently. There are places where this branch is better (or more up-to-date), but other cases where I prefer the approach in `swift-2.0`.

One of us should merge in improvements from the other branch. You or me, I'm happy either way!

**Comment by neilpa** [comment]
D'oh, I didn't look beyond open PRs before starting these changes. Otherwise I would've built on top of your existing branch. I'll rebase this against the existing branch later today.

**Comment by neilpa** [comment]
@jpsim Merged your changes into here as well.

**Comment by jpsim** [comment]
> Merged your changes into here as well.

Thanks! Could you change `.travis.yml` to point to `osx_image: xcode7` so we can get CI back up?

File: Commandant/ArgumentParser.swift [hunk 2]
```diff
@@ -61,15 +61,19 @@ public final class ArgumentParser {
 	/// Initializes the generator from a simple list of command-line arguments.
 	public init(_ arguments: [String]) {
 		// The first instance of `--` terminates the option list.
-		let params = split(arguments, maxSplit: 1, allowEmptySlices: true) { $0 == "--" }
+		let params = arguments.split(1, allowEmptySlices: true) { $0 == "--" }
```
Comment (jpsim) [inline Commandant/ArgumentParser.swift]: This should actually be `split(2, ...` as explained in #30.

**Comment by jpsim** [comment]
Once CI is back up (and passing), this is good to go! Thanks @neilpa!

File: Commandant/ArgumentParser.swift (hunk 2 above)
Comment (neilpa) [inline Commandant/ArgumentParser.swift]: They must've changed it back because I didn't see any failures locally when running the tests.

Comment (jpsim) [inline Commandant/ArgumentParser.swift]: Oh, that's entirely possible.

**Comment by mdiep** [comment]
It looks like CI is failing because `Box` was removed from `.gitmodules`, but not from `Carthage/Checkouts/`.

**Comment by neilpa** [comment]
Lets try this again.

**Comment by neilpa** [comment]
Not sure why the tests aren't running on travis

```
  [Info] Collecting info for testables... (1642 ms)
  run-test CommandantTests.xctest (macosx10.11, logic-test)

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

The build has been terminated
```

**Comment by ikesyo** [comment]
It seems that we need [xctool 0.2.6](https://github.com/facebook/xctool/releases/tag/v0.2.6) and Quick/Quick#388.

File: Commandant.xcworkspace/xcshareddata/Commandant.xcscmblueprint [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      "DVTSourceControlWorkspaceBlueprintRemoteRepositorySystemKey" : "com.apple.dt.Xcode.sourcecontrol.Git",
+      "DVTSourceControlWorkspaceBlueprintRemoteRepositoryIdentifierKey" : "D23C0CEAADB77074FDA4459000068A7940EB7AD0"
+    },
+    {
+      "DVTSourceControlWorkspaceBlueprintRemoteRepositoryURLKey" : "https:\/\/github.com\/jspahrsummers\/xcconfigs.git",
+      "DVTSourceControlWorkspaceBlueprintRemoteRepositorySystemKey" : "com.apple.dt.Xcode.sourcecontrol.Git",
+      "DVTSourceControlWorkspaceBlueprintRemoteRepositoryIdentifierKey" : "E084C86B03F81D63323C9E7510697EA528A758C7"
+    }
+  ]
+}
```
Comment (jdhealy) [inline Commandant.xcworkspace/xcshareddata/Commandant.xcscmblueprint]: This `xcscmblueprint` file should probably be removed and [ignored](https://github.com/github/gitignore/pull/1547). :point_up_2:

**Comment by jdhealy** [comment]
Other little thing: two references to Box still in Xcode project: lines [65](https://github.com/Carthage/Commandant/blob/swift-2/Commandant.xcodeproj/project.pbxproj#L65) and [127](https://github.com/Carthage/Commandant/blob/swift-2/Commandant.xcodeproj/project.pbxproj#L127).

**Comment by ikesyo** [comment]
The test are passed now on this branch. :tada:

**Comment by jpsim** [comment]
Looks good to me!

**Comment by jpsim** [comment]
Actually, I think the Components.plist needs to be updated now that Box is now longer used.

File: Cartfile.private [hunk 4]
```diff
@@ -1,3 +1,3 @@
-github "Quick/Quick" ~> 0.3
-github "Quick/Nimble" ~> 0.4
-github "jspahrsummers/xcconfigs" ~> 0.8
+github "jspahrsummers/xcconfigs" "ec5753493605deed7358dec5f9260f503d3ed650"
+github "Quick/Quick" "4c19d1257813c44df9c3c56b296b2b11ca40d4a9"
```
Comment (mdiep) [inline Cartfile.private]: Can be set to Quick `~> 0.6`?

File: Cartfile.private [hunk 5]
```diff
@@ -1,3 +1,3 @@
-github "Quick/Quick" ~> 0.3
-github "Quick/Nimble" ~> 0.4
-github "jspahrsummers/xcconfigs" ~> 0.8
+github "jspahrsummers/xcconfigs" "ec5753493605deed7358dec5f9260f503d3ed650"
```
Comment (mdiep) [inline Cartfile.private]: Can this be set to a release?

File: Cartfile.private (hunk 4 above)
Comment (ikesyo) [inline Cartfile.private]: Quick/Quick#388 is not included in the latest release.

**Comment by mdiep** [comment]
This looks good to me apart from the concerns about the `Cartfile`.

File: Cartfile.private (hunk 5 above)
Comment (ikesyo) [inline Cartfile.private]: It seems that jspahrsummers/xcconfigs#43 is required (it isn't released yet).

**Comment by ikesyo** [comment]
> Actually, I think the Components.plist needs to be updated now that Box is now longer used.

Is that [the Carthage's one](https://github.com/Carthage/Carthage/blob/c82c3d1cb6f35a146793a914f1fae0e14d95ce0c/Source/carthage/Components.plist#L30-L35)?

**Comment by mdiep** [comment]
Since the upstream dependencies haven't had releases yet, I'm good to merge this. But we should be sure to get updated releases for those dependencies before doing a Commandant release.

**Comment by mdiep** [comment]
Thanks for all the hard work everyone! :sparkling_heart:

**Comment by jdhealy** [comment]
:clap:

**Pull request merged**